src/Datasource/EntityTrait.php

```diff
@@ -402,7 +429,7 @@ public function &get(string $field): mixed
             return $result;
         }
 
-        if (!$fieldIsPresent && $this->requireFieldPresence) {
+        if (!$fieldIsPresent && $requireFieldPresence) {
```

Comment:
https://github.com/dereuromark/cakephp-shim/blob/b7f735e12edfe94fbfeb3b1b5e00e44ab70d00b2/src/Model/Entity/GetTrait.php#L20
OrFail() usually gets non null return values back 

I guess this here can return null if present as key.
That might be a different pattern or naming.

Reply:
Maybe getExistingOrFail()?